packages/lesswrong/lib/collections/posts/custom_fields.ts

```diff
@@ -914,7 +915,12 @@ addFieldsDict(Posts, {
     viewableBy: ['guests'],
     graphQLtype: GraphQLJSON,
     resolver: async (document, args, { currentUser }) => {
-      return await Utils.getTableOfContentsData({document, version: null, currentUser});
+      try {
+        return await Utils.getTableOfContentsData({document, version: null, currentUser});
+      } catch(e) {
+        Sentry.captureException(e);
```

Comment:
Is this how we're basically supposed to handle all errors? (I don't know that I've actually done any explicit error handling, but didn't have this as part of my workflow)

Reply:
Sometimes. Sort-of. If you do put a try-catch block around something, this is what it should look like, but try-catch blocks should be used sparingly. The client-side equivalent of this sort of thing is `withErrorBoundary`.

If you do put a try-catch around something and expect that it might suppress an error, then `Sentry.captureException` is the way to make sure we eventually notice and investigate that error. (If we didn't have something like `Sentry.captureException`, this would be a software anti-pattern called exception swallowing, which is very bad.)

Try-catch blocks like this can make things better in cases where (1) they cover a lot of code, or risky code, and (2) crashes in that code would otherwise break more than the code is responsible for. The purpose is to limit the scope of the breakage, if there's a bug. In this case, without the try-catch block, a bug in ToC generation would break the whole post page, whereas with the try-catch block, you get a working post page without a table of contents, which is better.

(I noticed this while working on the ToC code, when I made a version that crashed. We haven't had any problems with it crashing in production, so this is speculative.)